[averageandyyy] iP

averageandyyy opened the pull request.

averageandyyy committed:
Remove Duke.java

averageandyyy committed:
Implement Bento Greeting and Exit

averageandyyy committed:
Modify README.md

averageandyyy committed:
Implement echo behaviour and exit behaviour on bye input

averageandyyy committed:
Implement adding of task behavior and listing task behaviour
Inputting list will make Bento list all tasks. Otherwise, Bento assumes input to be task and adds it to the list.

averageandyyy committed:
Add Task and Main class. Implement marking and unmarking behavior.
Main class as entry point into program. Task class to better help organize program flow.
Bento will mark/unmark tasks based on index provided. Will need to handle cases when invalid indexes are provided in the future.

averageandyyy committed:
Modify code to adhere with coding standards. Add try-catch blocks to account for out of bounds indexing and wrong inputs.
Catch errors where user might provide non-integer inputs or out of bounds task number.

averageandyyy committed:
Refactor codebase
Extract constants such as GREETING_MESSAGE. Extract methods such as printNoTaskMessage()

averageandyyy committed:
Add ToDo Class and refactor code

averageandyyy committed:
Implement Deadline and Event classes and refactor code

averageandyyy committed:
Refactor code and remove magic numbers

averageandyyy committed:
Add generate_expected.sh and runtest_bento.sh and modify input.txt
While generate_expected.sh helps to automatically generate EXPECTED.TXT, its contents must still be verified manually before use with runtest_bento.sh
input.txt should include mis-inputs in the future as the commands/inputs grow.

averageandyyy committed:
Update generate_expected.sh

averageandyyy committed:
Create generic exception classes

averageandyyy committed:
Implement Task and ToDo specific exceptions

averageandyyy committed:
Implement Deadline specific exception

averageandyyy committed:
Implement Event specific exception

averageandyyy committed:
Update marking exceptions

averageandyyy committed:
Update text parsing behaviours and exceptions

averageandyyy committed:
Update EXPECTED.TXT

averageandyyy committed:
Merge branch 'branch-Level-5'

rexkoh425 commented on src/main/java/Bento.java:
```diff
@@ -0,0 +1,294 @@
+
+
+    // Data
+    private final Scanner in = new Scanner(System.in);
```
Could consider using capitalised character for this variable.

rexkoh425 commented on src/main/java/Bento.java:
```diff
@@ -0,0 +1,294 @@
+    // Event Functions
+    public void addEvent(String input) {
+        input = removeEventPrefix(input);
+        final int indexOfFrom = input.indexOf(FROM_PREFIX);
```
could use capitalised character since it is declared final

rexkoh425 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,37 @@
+        isDone = done;
+    }
+
+    public String getDoneMarker() {
```
Good job using get Method

rexkoh425 commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,10 @@
+        super(taskName);
+    }
+
+    @Override
```
Good job adding @override

rexkoh425 reviewed (commented):
LGTM. Can consider rechecking your variables that are declared final

averageandyyy committed:
Refactor code into packages

averageandyyy committed:
Merge branch 'branch-A-Packages'

averageandyyy committed:
Refactor code according to PR comments

averageandyyy committed:
Remove addTask method

averageandyyy committed:
Make Task class abstract and refactor code

averageandyyy committed:
Implement delete command

averageandyyy committed:
Implement task list saving behavior
- Able to load and save from ./data/save.txt
- Update tasks to return themselves as user commands
- Introduce new exceptions for when load and save processes encounter errors
- Will create new save.txt if not found
- Loading tasks makes use of handleUserInput which outputs success messages
- TODO: Create handleCommand functions that supress success messages

averageandyyy committed:
Merge branch 'branch-Level-6'

averageandyyy committed:
Merge branch 'branch-Level-7'

averageandyyy committed:
Update .gitignore to ignore jar directory

averageandyyy committed:
Update saveTaskList behavior
- Will now create a /data directory if not already created
- Introduced constants for readability and reusability
- Attempted and tested program with JAR file

averageandyyy committed:
Suppress success messages when loading from save.txt
- Introduce fromUserInput flag to determine whether to output success messages
- After loading, Bento will list all tasks

averageandyyy committed:
Remove /data directory from repository
- /data directory should only be present/created when the user uses Bento

averageandyyy committed:
Update .gitignore
- Future commits should disregard and /data/* files created when running Bento during development

averageandyyy committed:
Refactor code
- Introduce writeTaskStatus method in saveTaskList
- Introduce printMarkUpdate method in markTaskAsDone

averageandyyy committed:
Introduce additonal OOP classes
- Introduced abstract Command class with downstream classes overriding execute method
- Introduced Storage class to handle loading and saving of tasks
- Introduced TaskList class to manage user tasks
- Introduced Parser class to process user inputs as commands
- Introduced Ui class for I/O functionality

averageandyyy committed:
Add displayErrorMessage method for Ui

averageandyyy committed:
Refactor code
- Current behavior when save file is corrupted/not of the same format:
  - Bento will load the save file up till the point of corruption
  - Everything after the point of corruption is unfortunately lost

averageandyyy committed:
Merge branch 'branch-A-MoreOOP'

averageandyyy committed:
Modify Deadline and Event classes to take in LocalDate instead of Strings
- User must provide valid dates of the form dd/MM/yyyy in /by field for deadlines
- User must provide valid dates of the same form in /from and /to field for events
- Exceptions will be thrown should the user not comply
- Saving and loading mechanism updated to work with changes

averageandyyy committed:
Add show command to display deadlines/events associated with date of interest
- Usage: show dd/MM/yyyy
- Bento currently throws a default exception when user provides the wrong date
- Bento will tell user if no tasks associated with the provided date of interest is found

averageandyyy committed:
Fix DEFAULT_MESSAGE in BentoException

averageandyyy committed:
Add custom InvalidDateFormatException

averageandyyy committed:
Add new constants for better reusability

averageandyyy committed:
Introduce new constants

averageandyyy committed:
Merge pull request #1 from averageandyyy/branch-Level-8

Implement LocalDate functionality

averageandyyy committed:
Implement FindCommand
- Bento will search for tasks pertaining to the user's search phrase and list them
- Bento will throw an exception if it is unable to find any tasks of interest

averageandyyy committed:
Merge pull request #2 from averageandyyy/branch-Level-9

Implement FindCommand

averageandyyy committed:
Add JavaDoc comments for Command, Storage, Parser, Ui and Task-related classes

averageandyyy committed:
Add JavaDoc comments for Exception-related classes

averageandyyy committed:
Add JavaDoc comments for Command-related classes

averageandyyy committed:
Add JavaDoc comments for Constants-related classes

averageandyyy committed:
Add JavaDoc comments for remaining classes

averageandyyy committed:
Merge pull request #3 from averageandyyy/branch-A-JavaDoc

Merge branch-A-JavaDoc

averageandyyy committed:
Update test scripts

thoriritu commented:
Overall, the documentation of code through the use of comment is appropriately done. Re-check the final variable usage.

Good job!

thoriritu commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,131 @@
+     * @param parser The Parser instance to interpret the commands.
+     * @throws SaveFileErrorException if the file cannot be written or an error occurs during saving.
+     */
+    public void saveTaskList(TaskList tasks, Ui ui, Parser parser) throws SaveFileErrorException {
```
The saveTaskList and saveTaskListQuiet methods contain almost identical logic. You can try to extract the common logic into a private method and pass a boolean to indicate whether to print the success message.

thoriritu commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,88 @@
+     *
+     * @return The done marker as a string.
+     */
+    public String getDoneMarker() {
```
The getDoneMarker() method can be simplified using a ternary operator to improve readability and reduce verbosity.

thoriritu commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,74 @@
+     * @param index The index of the task to retrieve.
+     * @return The task at the specified index.
+     */
+    public Task retrieveTask(int index) {
```
The method retrieveTask could be renamed to getTask for consistency with other typical naming conventions (like getTaskCount), which would improve the readability of the code.

averageandyyy committed:
Update docs/README.md

averageandyyy committed:
Merge branch 'branch-A-UserGuide'

averageandyyy committed:
Create _config.yml

- Add Jekyll theme

averageandyyy committed:
Update _config.yml

averageandyyy committed:
Refactor code according to PR comments
- saveTaskList accepts a boolean isQuiet as a flag for printing success messages or not
- Bento will also congratulate the user if they have no tasks or have completed all tasks

averageandyyy committed:
Update code according to checkstyle

averageandyyy committed:
Update README.md to reflect auto-saving functionality
